lib/src/protocol_instance.rs

```diff
@@ -90,8 +90,11 @@ impl From<(&GuestInput, &Header, B256, &BlockProposedV2)> for BlockMetadataV2 {
             extraData: bytes_to_bytes32(&header.extra_data).into(),
 
             anchorBlockId: input.taiko.l1_header.number,
-            anchorBlockHash: input.taiko.l1_header.hash_slow(),
-
+            anchorBlockHash: if input.chain_spec.network() == Network::TaikoA7.to_string() {
```

Comment:
Should we compare the fork height instead of network name?

Reply:
it is inside BlockMetadataV2, so itself knows the fork.
The root cause of this hotfix is the holesky L1 has different hash algo (pectra fork introduced) than our current reth impl. We need a formal patch @johntaiko in the future if the L1 mainnet upgrades to that fork, then we can remove this workaround.